src/Compilers/CSharp/Portable/Binder/Binder_TupleOperators.cs

```diff
@@ -0,0 +1,369 @@
+            BoundExpression convertedRight = GenerateConversionForAssignment(operators.RightConvertedType, right, diagnostics);
+
+            TypeSymbol resultType = GetSpecialType(SpecialType.System_Boolean, diagnostics, node);
+            return new BoundTupleBinaryOperator(node, convertedLeft, convertedRight, kind, operators, resultType);
+        }
+
+        private TupleBinaryOperatorInfo BindTupleBinaryOperatorInfo(BinaryExpressionSyntax node, BinaryOperatorKind kind,
+            BoundExpression left, BoundExpression right, DiagnosticBag diagnostics)
+        {
+            TypeSymbol leftType = left.Type;
+            TypeSymbol rightType = right.Type;
+            int? leftCardinality = GetTupleCardinality(left);
+            int? rightCardinality = GetTupleCardinality(right);
+
+            if (leftCardinality.HasValue && rightCardinality.HasValue)
+            {
+                if (leftCardinality.Value != rightCardinality.Value)
+                {
+                    Error(diagnostics, ErrorCode.ERR_TupleSizesMismatchForBinOps, node, leftCardinality, rightCardinality);
+
+                    return new TupleBinaryOperatorInfo.Single(leftType ?? CreateErrorType(), rightType ?? CreateErrorType(),
+                        BinaryOperatorKind.Error, methodSymbolOpt: null, boolConversion: Conversion.NoConversion, boolOperator: default);
+                }
+
+                // typeless tuple literals are not nullable
+                bool leftNullable = leftType?.IsNullableType() == true;
+                bool rightNullable = rightType?.IsNullableType() == true;
+
+                return BindTupleBinaryOperatorNestedInfo(node, kind,
+                    GetTupleArgumentsOrPlaceholders(left), GetTupleArgumentsOrPlaceholders(right),
+                    leftNullable || rightNullable, diagnostics);
+            }
+
+            return BindTupleBinaryOperatorSingleInfo(node, kind, left, right, diagnostics);
+        }
+
+        private TupleBinaryOperatorInfo BindTupleBinaryOperatorSingleInfo(BinaryExpressionSyntax node, BinaryOperatorKind kind,
+            BoundExpression left, BoundExpression right, DiagnosticBag diagnostics)
+        {
+            TypeSymbol leftType = left.Type;
+            TypeSymbol rightType = right.Type;
+
+            if ((object)leftType != null && leftType.IsDynamic() || (object)rightType != null && rightType.IsDynamic())
+            {
+                return BindTupleDynamicBinaryOperatorSingleInfo(node, kind, left, right, diagnostics);
+            }
+
+            LookupResultKind resultKind;
+            BinaryOperatorAnalysisResult best = this.BinaryOperatorOverloadResolution(kind, left, right, node, diagnostics, out resultKind, out var _);
+
+            MethodSymbol resultMethod = null;
+            BinaryOperatorKind resultOperatorKind;
+            TypeSymbol convertedLeftType = null;
+            TypeSymbol convertedRightType = null;
```

Comment:
Initializers can be removed.

Reply:
That didn't work. If you go into the first branch of the `if (!best.HasValue)` then reach the `if (convertedLeftType is null || convertedRightType is null)` checks.

---
In reply to: [167973484](https://github.com/dotnet/roslyn/pull/23356#discussion_r167973484) [](ancestors = 167973484)